frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/validations/common-template-validations/common-templates-validations.ts

```diff
@@ -45,15 +37,24 @@ export const getTemplateValidations = (
       ? iUserTemplates.find((template) => iGetName(template) === userTemplateName)
       : null;
 
+  if (iUserTemplate) {
+    const validations = getValidationsFromTemplates(List([iUserTemplate]));
+    return getFieldValidations(validations, IDToJsonPath[fieldId]);
+  }
+
+  // Get OS, workload-profile and flavor attributes
+  const os = iGetVmSettingAttribute(state, id, VMSettingsField.OPERATING_SYSTEM);
+  const flavor = iGetVmSettingAttribute(state, id, VMSettingsField.FLAVOR);
+  const workloadProfile = iGetVmSettingAttribute(state, id, VMSettingsField.WORKLOAD_PROFILE);
+
+  // Get all the templates from common-templates
+  const commonTemplates = iGetLoadedCommonData(state, id, VMWizardProps.commonTemplates);
+
   // Get all the validations from the relevant templates:
   // Get the validations from the user template, if chosen.
   // If not, get the validations from Common-Templates based on OS, Workload-Profile and Flavor
-  const validations = iUserTemplate
-    ? JSON.parse(iGetIn(iUserTemplate, ['metadata', 'annotations', 'validations']))
-    : getValidationsFromTemplates(
-        // Get templates based on OS, Workload-Profile and Flavor
-        getRelevantTemplates(commonTemplates, os, workloadProfile, flavor),
-      );
+  const templates = getRelevantTemplates(commonTemplates, os, workloadProfile, flavor);
+  const validations = getValidationsFromTemplates(templates);
```

Comment:
I think there is an issue with the multiple templates validations.

IMO the validations should be ignored when one template does not have any validation for min/max/between. 
E.g. when workload profile is not selected, at this point it is not clear which single template will be selected. We should account for a possibility that each template can be selected and thus not limit the validations.

The exact validation for the min/max or none will be run again when the exact template is selected.

Reply:
> IMO the validations should be ignored when one template does not have any validation for min/max/between.

Yes, if we have one template, current code ignore validation if no validations are found.
If we have multiple templates we ignore the templates with no validations, but do use the validations we find in other templates.

> when workload profile is not selected, at this point it is not clear which single template will be selected. We should account for a possibility that each template can be selected and thus not limit the validations.

Correct, we where more restrictive, but not by much :-) in current code if we have multiple rules we take the most inclusive path possible.
For example:
if we have one template that require `min=3` and another that require `min=5`, we will take all values that pass at least one of this rule, in this case, we will end up with a `min=3` rule.

> The exact validation for the min/max or none will be run again when the exact template is selected.

Yes, in current code, we recheck on each change of os, load or flavor.

Reply:
yes - it works for templates with diffent validations.
But still I think we should support this use case:

template-1 : no validations
template-2 : min 2G memory validation

-> user selects 2G because s/he thinks this is the lowest value possible
->  user selects all other fields
only template-1 is available now
any memory value is possible

-> user creates VM on false assumption with 2G memory

Reply:
hmm ...
nice, will look how to fix that !